### PR TITLE
Check if a initialValue isn't null, undefined or a empty string

### PR DESCRIPTION
**Pull request by LauraBeatris**
Refer to issue: https://github.com/jungsoft/formup/issues/11

### CODE REVIEW

File: packages/core/src/utils/checkFormInputValue.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Checks if the value passed to FormInput is different than
+ * null, undefined or empty strings.
+ *
+ * Returns an boolean related to whether
+ * the field is empty or not
+ *
+ * @param value Value prop passed to FormInput
+ */
+const checkFormInputValue = (value: any) => { // eslint-disable-line
```
Comment (pedro-lb) [inline packages/core/src/utils/checkFormInputValue.ts:10]: No `eslint-disables`, `eslint` is there for a purpose and we should maintain it

Comment (pedro-lb) [inline packages/core/src/utils/checkFormInputValue.ts:10]: If this returns a `bool` we should use ts to indicate it

```suggestion
const checkFormInputValue = (value: any): boolean => { // eslint-disable-line
```

File: packages/core/src/utils/checkFormInputValue.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Checks if the value passed to FormInput is different than
+ * null, undefined or empty strings.
+ *
+ * Returns an boolean related to whether
```
Comment (pedro-lb) [inline packages/core/src/utils/checkFormInputValue.ts:5]: JSDoc has a marker for returns.

Add after `@params`:

```suggestion
 * @return {boolean} whether the field is empty or not
```

File: packages/core/src/contexts/FormGroupContext/FormGroupContainer.tsx [hunk 3]
```diff
@@ -39,7 +40,8 @@ const FormGroupContainer = ({
   ]);
 
   React.useEffect(() => {
-    if (initialValue) {
+    const isEmptyValue = checkFormInputValue(initialValue);
```
Comment (pedro-lb) [inline packages/core/src/contexts/FormGroupContext/FormGroupContainer.tsx:43]: `isEmptyValue` doesn't make sense here since the value is valid or not

**Comment by LauraBeatris** [comment]
Fixed in https://github.com/jungsoft/formup/pull/12/commits/6f2cebf270f0bb48489a1998d0da191f01916e15

**Pull request merged**